### DISCUSSION TITLE
Handling GitHub webhook retry

### DISCUSSION THREAD

**Discussion by nitinkcv** [community/community#24721] [Apps, API and Webhooks]
<p>Hi Team,</p>
<p>We’re planning to make use of GitHub Webhook for one of our use cases. One thing which is not clear to me is how the delivery failures are handled. For example:</p>
<ul>
<li>If the delivery failed for an event (say the downstream server was down), does GitHub retry the delivery again? If yes, what’s the frequency and max limit of the retry? I couldn’t get any details on this while searching numerous articles.</li>
<li>If GitHub doesn’t retry, are there api’s i can use to get the list of failed deliveries and then use an api to redeliver? I tried looking for it, but the closet i came to was <a href="https://api.github.com/hooks/%7BhookId%7D/deliveries" rel="nofollow noopener">https://github.com/hooks/{hookId}/deliveries</a> but that returns a “text/html” and not json for some weird reason. This is probably because it’s not an api per se.</li>
<li>Any other recommended way to get the failed deliveries and retry?</li>
</ul>
<p>Thanks,</p>
<p>Nitin</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>There are no retries of failed webhook deliveries and no API for listing or retrieving failed webhook events. If you need 100% guaranteed reliability, then you’ll need to use polling or some other system to find the events that happened while your system was unresponsive.</p>

**Comment by trungduyvu** [comment]
<p><a class="mention" href="/nitinkcv">@nitinkcv</a> I’m one of the founders of <a href="https://apitracker.com" rel="nofollow noopener">API Tracker</a> and this is one of the use case we solve for our customers. We let team set up a API proxy that forward to their webhook handler. You can enable auto retry, queue of failed calls, etc… with a click of a button</p>
<p>Sign up for a free account to give it a try or email me at <a href="mailto:trung@apitracker.com">trung@apitracker.com</a> if you have any question</p>

**Comment by octosteve** [comment]
<p>This might be helpful! <a href="https://github.blog/changelog/2021-06-30-webhook-deliveries-api/" class="inline-onebox" rel="noopener nofollow ugc">Webhook Deliveries API | GitHub Changelog</a></p>
<p>You can hit the deliveries endpoints for repos, orgs, and app events that have been delivered to your app.</p>
